src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```diff
@@ -0,0 +1,319 @@
+        case -1:
+          throw new Exception("Terminating pipeline on error as set in plugin configuration." + e.getMessage());
+        case 0:
+          return;
+        case 1:
+          emitter.emitError(new InvalidEntry<>(-1, e.getMessage(), record));
+          break;
+      }
+
+    }
+  }
+
+  @Override
+  public void destroy() {
+    super.destroy();
+    if (client != null) {
+      client.close();
+    }
+  }
+
+
+  /**
+   * Configures the <code>DlpSettings</code> to use user specified service account file or auto-detect.
+   *
+   * @return Instance of <code>DlpServiceSettings</code>
+   * @throws IOException thrown when there is issue reading service account file.
+   */
+  private DlpServiceSettings getSettings() throws IOException {
+    DlpServiceSettings.Builder builder = DlpServiceSettings.newBuilder();
+    if (config.getServiceAccountFilePath() != null) {
+      builder.setCredentialsProvider(() -> GCPUtils.loadServiceAccountCredentials(config.getServiceAccountFilePath()));
+    }
+    return builder.build();
+  }
+
+
+  /**
+   * Configuration object.
+   */
+  public static class Config extends GCPConfig {
+
+    @Macro
+    @Name("entire-record")
+    @Description("Check full record or a field")
+    private String entireRecord;
+
+    @Macro
+    @Name("field")
+    @Description("Name of field to be inspected")
+    @Nullable
+    private String field;
+
+    @Name("template-id")
+    @Description("ID of the Inspection Template defined in DLP")
+    private String templateId;
+
+    @Macro
+    @Name("on-error")
+    @Description("Error handling of record")
+    private String onError;
+
+    /**
+     * @return The name of field that needs to be inspected for sensitive data.
+     */
+    public String getFieldName() {
+      return field;
+    }
+
+    /**
+     * @return true if entire record to checked for sensitive data.
+     */
+    public boolean isEntireRecord() {
+      return entireRecord.equalsIgnoreCase("record");
+    }
+
+    /**
+     * @return -1 to stop processing, 0 to skip record, 1 to emit record to error.
+     */
+    public int onErrorHandling() {
+      if (onError.equalsIgnoreCase("stop-on-error")) {
+        return -1;
+      } else if (onError.equalsIgnoreCase("skip-record")) {
+        return 0;
+      } else {
+        return 1;
+      }
+    }
+
+
+    public void validate(FailureCollector collector, Schema inputSchema) {
+      if (!containsMacro("entire-record") && !isEntireRecord() && getFieldName() == null) {
+        collector.addFailure("Input type is specified as 'Field', " +
+                               "but a field name has not been specified.", "Specify the field name.")
+                 .withConfigProperty("field");
+      }
+
+      if (!isEntireRecord()) {
+        if (!containsMacro("field")) {
+          if (inputSchema.getField(getFieldName()) == null) {
+            collector.addFailure("Field specified is not present in the input schema", "")
```

Comment:
period at the end of the error message. Also what is the corrective action?

Reply:
Done